### PR TITLE
Improve parlai cold start

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Delay the import of a few very expensive imports.

**Testing steps**
Before:
```
$ time parlai -h
** fvcore version of PathManager will be deprecated soon. **
** Please migrate to the version in iopath repo. **
https://github.com/facebookresearch/iopath

usage: parlai [-h] [--helpall] [--version] COMMAND ...

       _
      /")
     //)
  ==//'=== ParlAI
   /

optional arguments:
  -h, --help               show this help message and exit
  --helpall                List all commands, including advanced ones.
  --version                Prints version info and exit.

Commands:

  eval_model (em, eval)    Evaluate a model
  display_data (dd)        Display data from a task
  display_model (dm)       Display model predictions.
  train_model (tm, train)  Train a model
  interactive (i)          Interactive chat with a model on the command line
  safe_interactive         Like interactive, but adds a safety filter
  self_chat                Generate self-chats of a model
parlai -h  7.02s user 4.20s system 36% cpu 30.978 total
```

After:
```
$ time parlai -h
usage: parlai [-h] [--helpall] [--version] COMMAND ...

       _
      /")
     //)
  ==//'=== ParlAI
   /

optional arguments:
  -h, --help                  show this help message and exit
  --helpall                   List all commands, including advanced ones.
  --version                   Prints version info and exit.

Commands:

  eval_model (em, eval)       Evaluate a model
  dump_data_to_conversations  Evaluate a model
  display_data (dd)           Display data from a task
  display_model (dm)          Display model predictions.
  train_model (tm, train)     Train a model
  interactive (i)             Interactive chat with a model on the command line
  safe_interactive            Like interactive, but adds a safety filter
  self_chat                   Generate self-chats of a model
parlai -h  0.87s user 0.54s system 11% cpu 12.440 total
```

### CODE REVIEW

File: parlai/core/metrics.py [hunk 1]
```diff
@@ -505,9 +493,13 @@ def compute_many(
         :return: (rouge-1, rouge-2, rouge-L)
         """
         # possible global initialization
-        global rouge
-        if rouge is None:
+        try:
+            import rouge
+        except ImportError:
+            # User doesn't have py-rouge installed, so we can't use it.
+            # We'll just turn off rouge computations
```
Comment (EricMichaelSmith) [inline parlai/core/metrics.py:500]: Hmm should we print a warning here? (Same with the bleu imports)

File: parlai/core/torch_generator_agent.py [hunk 2]
```diff
@@ -49,19 +49,6 @@
 )
 
 
-try:
```
Comment (EricMichaelSmith) [inline parlai/core/torch_generator_agent.py:52]: Yeah, strange that we imported these before

**Review by EricMichaelSmith — APPROVED** [review]
Cool - yeah, makes sense to not import until/if we need them

File: parlai/core/metrics.py (hunk 1 above)
Comment (stephenroller) [reply parlai/core/metrics.py:500]: Hmm, future diff

**Pull request merged**